### PR TITLE
chore: fix linting

### PR DESCRIPTION
**Pull request by ldez**
# Description

Inside the configuration, the option `new-from-rev` is defined, this option will filter the results to only display reports from the diff between you branch and `origin/develop`.

```yml
issues:
  new-from-rev: origin/develop # report only new issues with reference to develop branch
```

https://golangci-lint.run/welcome/faq/#why---new-from-rev-or---new-from-patch-dont-seem-to-be-working-in-some-cases

I guess this was used because the project didn't want to fix all the reports in one row.

The diff is done with the branch `develop` (which was the default branch of https://github.com/0xPolygon/polygon-edge) but your default branch is not the same (`prod`).
So you will have differences inside your CI depending on which diff is evaluated. 

This PR will all the reports, so it removes the need for `new-from-rev`

Also, the project depends on min go1.21 and not go1.20.
And I run `go mod tidy`.

`actions/setup-go` must be done after `actions/setup-checkout` for cache optimizations.

I also updated the version of the actions.

This PR targets the branch of the PR https://github.com/Hydra-Chain/hydragon-node/pull/70, so this PR will be merged inside another PR and not inside `prod`.

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [ ] I have assigned this PR to myself
- [ ] I have added at least 1 reviewer
- [ ] I have added the relevant labels
- [ ] I have updated the official documentation
- [ ] I have added sufficient documentation in code

## Testing

- [ ] I have tested this code with the official test suite
- [ ] I have tested this code manually

### CODE REVIEW

**Comment by ldez** [comment]
The failure of `Pull Request CI / Test / HydraGon (pull_request)` is related to a flaky test, not to my PR.
You can re-launch this job.

File: secrets/encryptedlocal/prompt.go [hunk 1]
```diff
@@ -15,10 +15,11 @@ import (
...
 var (
 	ErrInvalidPassword = errors.New(
-		"Password must contain at least one number, one uppercase letter, one special character, and be at least 8 characters long",
+		"password must contain at least one number, one uppercase letter, one special character," +
+			" and be at least 8 characters long",
 	)
-	ErrPasswordMismatch    = errors.New("Passwords do not match")
-	ErrTerminatedOperation = errors.New("Operation terminated")
+	ErrPasswordMismatch    = errors.New("passwords do not match")
+	ErrTerminatedOperation = errors.New("operation terminated")
```
Comment (Vitomir2) [inline secrets/encryptedlocal/prompt.go:22]: We might want to start the error messages with capital letter.

Comment (ldez) [reply secrets/encryptedlocal/prompt.go:22]: Your linters configuration says the opposite.
It's a Go convention: the first character should not be an upper case letter and the last character should not be a punctuation.

**Review by Vitomir2 — APPROVED** [review]

File: secrets/encryptedlocal/prompt.go (hunk 1 above)
Comment (Vitomir2) [reply secrets/encryptedlocal/prompt.go:22]: Yeah, I agree that the linter says the opposite. Thanks for marking this out, I will discuss it with the team and will decide if we will follow this convention.

**Comment by ldez** [comment]
The failure of `Pull Request CI / Test / HydraGon (pull_request)` is still related to flaky tests: it's not the same test that fails at each run.

**Pull request merged**